File: civitas/src/civitas/SorpresaIrCasilla.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SorpresaIrCasilla extends Sorpresa {
    
    private int valor;
    private Tablero tablero;
    
    SorpresaIrCasilla(Tablero tablero, int valor, String texto) {
        super(texto);
        this.tablero = tablero;
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            int casillaActual = todos.get(actual).getNumCasillaActual();
            int tirada = tablero.calcularTirada(casillaActual, valor);
            int nuevacasilla = tablero.nuevaPosicion(casillaActual, tirada);
            todos.get(actual).moverACasilla(nuevacasilla);
            tablero.getCasilla(nuevacasilla).recibeJugador(actual, todos);
        }
    }
    
    @Override
    public String toString() {
        String salida = "Valor: " + valor + "Tablero: " + tablero + 
                super.toString();
        return salida;
    }
    
}

File: civitasRuby/lib/sorpresa_ir_casilla.rb
# encoding:utf-8
module Civitas

  class SorpresaIrCasilla < Sorpresa
  
    def initialize(texto, tablero, valor)
      super(texto)
      @tablero = tablero
      @valor = valor
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        super(actual, todos)
        casilla_actual = todos.at(actual).num_casilla_actual
        tirada = @tablero.calcular_tirada(casilla_actual, @valor)
        nueva_casilla = @tablero.nueva_posicion(casilla_actual, tirada)
        todos.at(actual).mover_a_casilla(nueva_casilla)
        @tablero.get_casilla(nueva_casilla).recibe_jugador(actual, todos)
      end
    end
    
    def to_string
      return super.to_string + "\nValor: #{@valor} " 
    end
  
  public_class_method :new  
  end
end